### PR TITLE
Revenant Rework (NOW WITH LESS CLOSING)

### PR DESCRIPTION
**Pull request by ghost**
This PR will make revenants more focused around using the environment, as well as giving them more weaknesses such as salt and holy water.

### CODE REVIEW

**Comment by SconesC** [comment]
Someone spawned one yesterday and it was fucking atrocious, just a meme ghost that spammed mind spike on me and the rest of Medbay so we just packed heads and downed mannitol for like 30 minutes while it tried, and defibbed anyone it absorbed

I hope this fixes that

**Comment by Cheridan** [comment]
I read the title as "NOW WITH LESS CLOTHING"

:blush:

**Comment by AlexCrimson** [comment]
inb4Xhuis the PR destroyer strikes again

**Comment by Fox-McCloud** [comment]
As a heads up, the overload lighting ability will kill off one of your lives when you purchase it, unless your absorb essence first.

Likewise, you can't use overload lighting, at all, without killing off one of your lives unless...again, you've absorbed someone's essence first.

This pretty much means you're incapable of doing anything other than waiting around for people do die.

**Comment by Xhuis** [comment]
That's the point of revenant.

**Comment by Fox-McCloud** [comment]
major bug: when you die, you don't ghostize() and the now dead revenant isn't qdel'd either; you just go into the new animation, freeze for a bit, then you can go about doing whatever you want...indefnitely.

**Comment by Xhuis** [comment]
Not having that issue on my end, I'll see if I can reproduce it though.

**Comment by Xhuis** [comment]
@Fox-McCloud You might want to wait before porting revenants until after this PR is finished up.

**Comment by Fox-McCloud** [comment]
@Xhuis 

That was my intention, XHuis; it was just put up as a holder =)

File: code/modules/mob/living/simple_animal/revenant/revenant_spawn_event.dm [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#define REVENANT_SPAWN_THRESHOLD 20
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/revenant/revenant_spawn_event.dm]: seems a bit high.

Comment (Anonus) [inline code/modules/mob/living/simple_animal/revenant/revenant_spawn_event.dm]: 20 is a good chunk of most rounds population and is actually getting to the point where you wouldn't want to spawn one. Maybe 10?

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/revenant/revenant_spawn_event.dm]: The dead mob list includes ghosts and dead mobs on the station z-level. By the time this event occurs, there are usually enough dead mobs to make this happen.

**Comment by Xhuis** [comment]
Woo

**Comment by AlexCrimson** [comment]
:o

**Comment by Lumipharon** [comment]
Can revenants still communicate with the living? Because it's seriously meta/10 round breaking shit.

**Comment by Xhuis** [comment]
You know, you could have posted on the feedback thread rather than necroing a 3-monrh old PR.

**Comment by Lumipharon** [comment]
I unno, it popped up in the git notifications.

**Pull request merged**